### ISSUE TITLE
Mount() doesn't match root path of the sub-router

### ISSUE THREAD

**Issue by VojtechVitek** [go-chi/chi#260]
Given
```go
package main

import (
	"net/http"

	"github.com/go-chi/chi"
)

func main() {
	r := chi.NewRouter()
	r.Mount("/path", MyRouter())

	http.ListenAndServe(":9876", r)
}

func MyRouter() http.Handler {
	r := chi.NewRouter()

	r.Get("/*", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte("works..\n"))
	})

	return r
}
```

How come `localhost:9876/path/` is not matched and returns 404?
```
$ curl localhost:9876/path
works..

$ curl localhost:9876/path/
404 page not found

$ curl localhost:9876/path/hey
works..
```

Brought up in #259 by @hovhannest.

**Comment by olihey** [comment]
Any fix for this?
I just went into the exact same issue and the only way I could fix it was to use
```r.Mount("/path/", MyRouter())```
but then `localhost:9876/path` fails, of course 😢

**Comment by hovhannest** [comment]
I've changed the "thee.go" file line 444 from
if len(xsearch) == 0 {
to
if len(xsearch) == 0  || xsearch == "/" {

and I'm using r.Mount("/path", MyRouter())

I've suggested this change, but my pull request was denied. May be I'm doing smt. wrong, but it's working for me.

**Comment by inancgumus** [comment]
Why is that your request has been denied?

**Comment by pkieltyka** [comment]
I’m away on vacation but I will revisit this issue when I return

**Comment by VojtechVitek** [comment]
@inancgumus you can see yourself in #259. Check Travis to see why.

**Comment by pkieltyka** [comment]
Fixed in https://github.com/go-chi/chi/pull/281

**Comment by lrstanley** [comment]
Any idea on user pushback for merging #281 into master? I've had to use the **fix-260** branch for over 5 projects at this time. Haven't ran into any issues, and I personally find that the implementation makes sense. When mounting, one would think that it's treated the same way that the primary/first `/` is handled.

**Comment by pkieltyka** [comment]
I merged the #281 PR which fixes this. Give master a try and let us know
